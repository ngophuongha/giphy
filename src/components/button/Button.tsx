import { ButtonProps } from "./Button.type";

export const Button = ({ btnTitle, ...props }: ButtonProps): JSX.Element => {
  //TODO: Add className for button variants
  return <button {...props}>{btnTitle}</button>;
};
