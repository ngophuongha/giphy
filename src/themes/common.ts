import { DefaultTheme } from "styled-components";

const screens = {
  xs: "320px",
  sm: "640px",
  md: "768px",
  lg: "1024px",
  xl: "1280px",
  xxl: "1536px",
};

const commonColors = {
  transparent: "transparent",
  light: "#fff",
  dark: "#171717",
  toggleBackground: "#444444",
  toggleIcon: "#ffa701",
  primary: "#444444",
};

const common: DefaultTheme = {
  colors: commonColors,
};

export { common, screens, commonColors };
