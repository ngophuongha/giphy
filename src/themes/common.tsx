import { DefaultTheme } from "styled-components";

const screens = {
  xs: "320px",
  sm: "640px",
  md: "768px",
  lg: "1024px",
  xl: "1280px",
  xxl: "1536px",
};

const commonColors = {
  transparent: "transparent",
  light: "#fff",
  dark: "#1b1c1c",
  toggleBackground: "#444444",
  toggleIcon: "#ffa701",
  primary: "#2b5dab",
  secondary: "a1343e",
  brickRed: "#a1343e",
  coralBlue: "#2b5dab",
};

const common: DefaultTheme = {
  colors: commonColors,
};

export { common, screens, commonColors };
