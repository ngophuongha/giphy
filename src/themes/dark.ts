import { DefaultTheme } from "styled-components";

const darkColors = {
  textColor: "#ffffff",
  btnTextColor: "#1b1c1c",
  background: "#1b1c1c",
  borderColor: "#ffffff",
  hoverBackground: "#ffffff",
  hoverTextColor: "#444444",
  primary: "#2b5dab",
  hoverPrimary: "#25539c",
  secondary: "#a1343e",
  hoverSecondary: "#A41d2a",
  default: "#1b1c1c",
  hoverDefault: "#1b1c1c",
  formHeader: "#ffffff",
};

const dark: DefaultTheme = {
  colors: darkColors,
};

export { dark, darkColors };
