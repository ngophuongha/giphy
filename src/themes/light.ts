import { DefaultTheme } from "styled-components";

const lightColors = {
  transparent: "transparent",
  light: "#fff",
  dark: "#d2d2d2",
  textColor: "#000000",
  background: "#fff",
  borderColor: "#d3d3d3",
  hoverBackground: "#444444",
  hoverTextColor: "#ffffff",
  primary: "#fff",
  secondary: "#fff",
  default: "#fff",
};

const light: DefaultTheme = {
  colors: lightColors,
};

export { light, lightColors };
