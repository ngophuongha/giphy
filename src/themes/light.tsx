import { DefaultTheme } from "styled-components";

const lightColors = {
  transparent: "transparent",
  light: "#fff",
  dark: "#d2d2d2",
  textColor: "#000000",
  btnTextColor: "#fff",
  background: "#fff",
  borderColor: "#d3d3d3",
  hoverBackground: "#444444",
  hoverTextColor: "#ffffff",
  primary: "#2b5dab",
  hoverPrimary: "#25539c",
  secondary: "#a1343e",
  hoverSecondary: "#A41d2a",
  default: "#fff",
  hoverDefault: "#1b1c1c",
  formHeader: "#2b5dab",
};

const light: DefaultTheme = {
  colors: lightColors,
};

export { light, lightColors };
