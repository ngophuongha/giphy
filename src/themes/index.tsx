import { FC } from "react";
import { ThemeProvider } from "styled-components";
import { useAppSelector } from "../store/hooks";
import { dark } from "./dark";
import { light } from "./light";
import { ThemeType } from "../store/theme/theme";

export * from "./common";
export * from "./light";
export * from "./dark";

export const StyledThemeProvider: FC = ({ children }) => {
  const currentTheme = useAppSelector((state) => state.theme.theme);
  return (
    <ThemeProvider theme={currentTheme === ThemeType.Dark ? dark : light}>
      {children}
    </ThemeProvider>
  );
};
