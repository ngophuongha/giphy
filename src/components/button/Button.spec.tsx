import { render, screen, fireEvent } from "../../../test";
import { lightColors } from "../../themes";
import { StyledButton as Button } from "./Button.styled";

describe("Styled Button", () => {
  it("should match snapshot", () => {
    const { container } = render(<Button btnTitle={"Default Btn"} />);
    expect(container).toMatchSnapshot();
  });

  it("should have correct variant", () => {
    render(<Button btnTitle={"Primary Btn"} variant="primary" />);

    const primaryBtn = screen.getByRole("button");
    expect(primaryBtn).toHaveStyle(`background-color: ${lightColors.primary}`);

    // fireEvent.mouseEnter(primaryBtn);
    // screen.debug(primaryBtn);
    // expect(primaryBtn).toHaveStyle(
    //   `background-color: ${lightColors.hoverPrimary}`
    // );
  });
});
