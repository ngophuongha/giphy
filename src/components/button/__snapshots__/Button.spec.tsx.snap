// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Button should match snapshot 1`] = `
.c0 {
  padding: 0.25rem !important;
  border-radius: 0.25rem;
  border: 1px solid #fff;
  color: #fff;
  background-color: #fff;
}

.c0:hover {
  background-color: #1b1c1c;
  color: #fff;
}

<div>
  <button
    class="c0"
  >
    Default Btn
  </button>
</div>
`;
